fix: Export `DiagnosticVisitor` from execution/mod.rs

janis-me opened the pull request:
Exporting the `DiagnosticVisitor` as mentioned in #40 - so it can be used with `use depends::{DiagnosticVisitor};`. I hope exporting it was intended.

I tested this by patching the depends package locally with

```toml
[patch.crates-io]
depends = { git="https://github.com/janis-me/depends-rs", branch="fix/export-diagnostics-visitor" }
```
and it works well!

~ Cheers

janis-me committed:
fix: Export `DiagnosticVisitor` from execution/mod.rs

Justice4Joffrey commented:
@janis-me thanks for the PR. This type is exported with the feature "test-utils", but I think that's quite unclear. Please remove the existing export in `test_utils.rs` and I'd be happy to merge this.

janis-me committed:
chore: Remove `DiagnosticVisitor` export from test_utils,

Another commit follows that fixes the `raising_the_stakes` test with updated exports

janis-me committed:
fix: adjust `raising_the_stakes` test to import `DiagnosticVisitor` from `depends`, not `depends::test_utils`

janis-me committed:
fix: remove redundant import of `DiagnosticVisitor`

janis-me commented:
Got it! Could have seen the import in the `test_resolve_graph`, sorry :)

Updated the PR to remove the export and update the test in `raising_the_stakes.rs`
